ClubDoorman/BadMessageManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Security.Cryptography;
+using System.Text;
+
+namespace ClubDoorman;
+
+public class BadMessageManager
+{
+    private const string Path = "data/bad-messages.txt";
+    private readonly SemaphoreSlim _semaphore = new(1);
+    private readonly HashSet<string> _bad = File.ReadAllLines(Path).ToHashSet();
+
+    public bool Bad(string message) => _bad.Contains(ComputeHash(message));
+
+    public async ValueTask MarkAsBad(string message)
+    {
+		if string.IsNullOrEmpty(message)
+		{
+			return;
+		}
+        var hash = ComputeHash(message);
+        if (_bad.Add(hash))
+        {
+            using var token = await SemaphoreHelper.AwaitAsync(_semaphore);
+            await File.AppendAllLinesAsync(Path, [hash]);
+        }
+    }
+
+    private static string ComputeHash(string message)
+    {
+        var bytes = Encoding.UTF8.GetBytes(message);
+        using (var hasher = SHA512.Create())

```

Comment:
Instead of creating new instance each time you can use HashData static
https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.sha512.hashdata?view=net-8.0